### ISSUE TITLE
vim-stay in neovim produces E484 error when opening new file

### ISSUE THREAD

**Issue by zhimsel** [zhimsel/vim-stay#31]
If I open a file that doesn't already have a viewdir file (i.e. that vim-stay has not seen before), it produces an error stating that it can't open that file. This is expected, since it doesn't exist until the buffer is saved, and the viewdir file is created.

For example. I created a blank file, `/tmp/test_file`, and after confirming that there is no viewdir file for it already (clearing the viewdir), I open it in neovim: `nvim /tmp/test_file`. This produces the following error in the statusline message section:

```
vim-stay could not read the view session file! Vim error E484: Can't open file /Users/zhimsel/.local/share/nvim/view/=+private=+tmp=+test_file=
```

This is only present in neovim, not vim. Once the buffer is written, `:w`, a viewdir file is created in `~/.local/share/nvim/view` for that file, and any subsequent times neovim opens that file, the error is not produced (since it exists). 

I have a feeling it's related to [this section](https://github.com/kopischke/vim-stay/blob/master/plugin/stay.vim#L53-L69), but I don't know `vim-stay` or vimscript well enough to troubleshoot.  

I also noticed that there's a bunch of [version specific conditionals](https://github.com/kopischke/vim-stay/blob/master/autoload/stay/shim.vim) that may be related as well.

Any ideas?

---

```
:version
NVIM 0.1.4
Build type: RelWithDebInfo
Compilation: /usr/local/Library/ENV/4.3/clang -Wconversion -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DDISABLE_LOG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-p
rototypes -std=gnu99 -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -DHAVE_CONFIG_H -I/tmp/neovim-20160501-1882-n124ji/neovim
-0.1.4/build/config -I/tmp/neovim-20160501-1882-n124ji/neovim-0.1.4/src -I/tmp/neovim-20160501-1882-n124ji/neovim-0.1.4/deps-build/usr/include -I/tmp/neovim-20160501-1
882-n124ji/neovim-0.1.4/deps-build/usr/include -I/tmp/neovim-20160501-1882-n124ji/neovim-0.1.4/deps-build/usr/include/luajit-2.0 -I/tmp/neovim-20160501-1882-n124ji/neo
vim-0.1.4/deps-build/usr/include -I/tmp/neovim-20160501-1882-n124ji/neovim-0.1.4/deps-build/usr/include -I/tmp/neovim-20160501-1882-n124ji/neovim-0.1.4/deps-build/usr/
include -I/tmp/neovim-20160501-1882-n124ji/neovim-0.1.4/deps-build/usr/include -I/usr/local/opt/gettext/include -I/usr/include -I/usr/include -I/tmp/neovim-20160501-18
82-n124ji/neovim-0.1.4/build/src/nvim/auto -I/tmp/neovim-20160501-1882-n124ji/neovim-0.1.4/build/include
Compiled by zach@randir

Optional features included (+) or not (-): +acl   +iconv    +jemalloc
For differences from Vim, see :help vim-differences

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/local/Cellar/neovim/0.1.4/share/nvim"
```

**Comment by zhimsel** [comment]
Any updates on this? Still happening on latest master. Can I be of any help?

**Comment by kopischke** [comment]
I have a fix for that in the [release/1.4.0](https://github.com/kopischke/vim-stay/tree/release/1.4.0) branch – you could give it a whirl. However, at this point, I cannot in good conscience give an ETA for that branch to be merged into master.

**Comment by zhimsel** [comment]
> I have a fix for that in the release/1.4.0 branch – you could give it a whirl.

Great, thanks! I'll test out that branch.

> However, at this point, I cannot in good conscience give an ETA for that branch to be merged into master.

Totally understandable.

**Comment by zhimsel** [comment]
Huh... Didn't realize I was [already using the `release/1.4.0` branch](https://github.com/zhimsel/dotfiles/blob/b92e3242104b53bbc398f6485a8f424220b8f7ff/vim/vimrc#L52):
https://github.com/kopischke/vim-stay/compare/release/1.4.0...zhimsel:master

Could you point me to the commit that fixes this issue?

**Comment by kopischke** [comment]
My bad, looks like I never pushed that yet …

**Comment by zhimsel** [comment]
Hey, any chance of getting that fix pushed to the `release/1.4.0` branch?

**Comment by kopischke** [comment]
The honest answer is no: I have come to the point I had to admit to myself that, as far as I’m concerned, my Vim plug-ins are EOL. I don’t do much coding anymore, and when I do, I barely ever use Vim or its derivatives (heresy, I know). [My _vim-fetch_ plug-in has found a new maintainer in the person of Wang Shidong](https://github.com/wsdjeg/vim-fetch), and I would gladly divest myself of _vim-stay_ too, if anybody was interested … you don’t happen to want take over, would you 😉?

**Comment by zhimsel** [comment]
> you don’t happen to want take over, would you 😉?

Hm. I mean, I wouldn't necessarily be opposed to it, but I don't know vimscript well enough (or your codebase here) to be confident in my ability. I'd have to take a good look through the whole of `vim-stay` before I'd say yes.

**Comment by zhimsel** [comment]
If you wouldn't mind giving me a brief walkthrough of the code, I'm a quick learner and probably would be able to pick it up easily.

**Comment by kopischke** [comment]
>If you wouldn't mind giving me a brief walkthrough of the code

I’ll gladly do that; we have a three day weekend upcoming in Germany, so I should find the time to do it, especially considering _vim-stay_ is not a very complex plug-in (it’s essentially a wrapper around `mkview` and `loadview` that irons out the kinks and idiosyncrasies of those two commands).

**Comment by zhimsel** [comment]
If it's this coming weekend, I should be able to find some time to link up. Shoot me an email at zach@himsel.net and we can coordinate that.
